features/net/network-socket/nsapi_types.h

```diff
@@ -148,6 +148,18 @@ typedef enum nsapi_option {
 } nsapi_option_t;
 
 
+/*  C linkage definition
+ *
+ *  Attribute garuntees emitted function has C linkage.
+ *  Available in both C and C++
+ */
+#ifdef __cplusplus
+#define NSAPI_C_LINKAGE extern "C"
```

Comment:
right - my point was why a macro and not extern "C" in the cpp files? The macro seems unnecessary.